vaadin-spring/src/main/java/com/vaadin/flow/spring/ForwardingRequestWrapper.java

```diff
@@ -42,24 +42,15 @@ public ForwardingRequestWrapper(HttpServletRequest request) {
 
     @Override
     public String getServletPath() {
-        String pathInfo = super.getPathInfo();
-        if (pathInfo == null) {
-            // the path where a ServletForwardingController is registered is not
-            // a real servlet path
-            return "";
-        } else {
-            return super.getServletPath();
-        }
+        return (super.getPathInfo() == null) ? super.getServletPath() : "";
```

Comment:
The proposed code seems to invert the original logic

Reply:
Right. Yet another case for my claim that readability trumps conciseness.